### ISSUE TITLE
Add `forecast` method

### ISSUE THREAD

**Issue by jwdink** [strongio/torch-kalman#10]
Currently,`_filter` and `forward` methods are written with a training loop in mind: the dimensions of their outputs match the dimensions of the inputs, so it's easy to plug the two into a loss function.

But obviously in practice, need a way to get predictions out to some horizon.

This method should probably handle lagging of inputs to nn-modules that supplement the predictions of the kalman-filter. 

One tricky thing about that is some need to be lagged but some shouldn't. E.g., 

- *Shouldn't be lagged:* The components of a fourier series, weather/forecasts.
- *Should be lagged:* S&P 500, customer activity up to this point.
